app/assets/javascripts/discourse/app/components/d-editor.hbs

```diff
@@ -21,6 +21,8 @@
                   @options={{hash icon=b.icon focusAfterOnChange=false}}
                   class={{b.className}}
                 />
+              {{else if b.component}}
```

Comment:
allows to render a dedicated component in the toolbar, in this case the  `<EmojiPicker />`

Reply:
Can we use a different approach? The AI plugin uses the existing API:

https://github.com/discourse/discourse-ai/blob/11d0f60f1ec72a882067c65d0b62c02760f68840/assets/javascripts/initializers/ai-helper.js#L65-L74

I don't love the `querySelector`, but it avoids opening the gates for _any_ `component` to be rendered there